src/utils/transaction.ts

```diff
@@ -141,6 +154,101 @@ export const makeTxConfigFromSignedTx = (
   return txConfig;
 };
 
+// needs testing
+export const makeTxConfigFromTransactionResponse = (
+  decodedTx: TransactionResponse,
+  assets: ExtendedAsset[],
+  network: Network,
+  accounts: StoreAccount[]
+): ITxConfig => {
+  const contractAsset = getAssetByContractAndNetwork(decodedTx.to || undefined, network)(assets);
+  const baseAsset = getBaseAssetByNetwork({
+    network,
+    assets
+  })!;
+
+  const txConfig = {
+    rawTransaction: {
+      to: decodedTx.to as TAddress,
+      value: hexlify(decodedTx.value),
+      gasLimit: hexlify(decodedTx.gasLimit),
+      data: decodedTx.data,
+      gasPrice: hexlify(decodedTx.gasPrice),
+      nonce: hexlify(decodedTx.nonce),
+      chainId: decodedTx.chainId,
+      from: decodedTx.from as TAddress
+    },
+    receiverAddress: (contractAsset
+      ? decodeTransfer(decodedTx.data)._to
+      : decodedTx.to) as TAddress,
+    amount: contractAsset
+      ? fromTokenBase(toWei(decodeTransfer(decodedTx.data)._value, 0), contractAsset.decimal)
+      : formatEther(decodedTx.value),
+    network,
+    value: toWei(decodedTx.value.toString(), getDecimalFromEtherUnit('ether')).toString(),
+    asset: contractAsset || baseAsset,
+    baseAsset,
+    senderAccount: getStoreAccount(accounts)(decodedTx.from as TAddress, network.id)!,
+    gasPrice: decodedTx.gasPrice.toString(),
+    gasLimit: decodedTx.gasLimit.toString(),
+    data: decodedTx.data,
+    nonce: decodedTx.nonce.toString(),
+    from: decodedTx.from as TAddress
+  };
+  return txConfig;
+};
+
+export const makeTxConfigFromTxReceipt = (
+  txReceipt: ITxReceipt,
+  assets: ExtendedAsset[],
+  networks: Network[],
+  accounts: StoreAccount[]
+): ITxConfig => {
+  const networkDetected = getNetworkById(txReceipt.asset.networkId, networks);
+  const contractAsset = getAssetByContractAndNetwork(
+    txReceipt.to || undefined,
+    networkDetected
+  )(assets);
+  const baseAsset = getBaseAssetByNetwork({
+    network: networkDetected || ({} as Network),
+    assets
+  });
+
+  const txConfig = {
+    rawTransaction: {
+      to: txReceipt.to,
+      value: bigNumberify(txReceipt.value).toHexString(),
```

Comment:
Prefer `bigify`.
Applies for the entire transformation

Reply:
Our current TX types use `BigNumber` from ethers.js. Hence why I use it. I would prefer it we changed this in another PR.